py-polars/polars/series/datetime.py

```diff
@@ -80,15 +83,18 @@ def median(self) -> dt.date | dt.datetime | dt.timedelta | None:
 
         """
         s = wrap_s(self._s)
-        out = s.median()
+        out = s._s.median()
         if out is not None:
             if s.dtype == Date:
-                return _to_python_date(int(out))
-            else:
+                # 86_400_000_000 = microseconds in days
+                return _to_python_datetime(int(out * 86_400_000_000), "us")
```

Comment:
same, is there a constant on the python side?

Reply:
We have `NS_PER_SECOND` AND `SECONDS_PER_DAY` so I brought those in to create a `US_PER_DAY`. Thanks for the suggestions.